### ISSUE TITLE
Rust 1.26.1 point release

### ISSUE THREAD

**Issue by pietroalbini** [rust-lang/rust#50756]
> **Note:** there is no decision on whether 1.26.1 should be released yet. This is just a wishlist.

There are a few bad bugs in the 1.26.0 release, and there is a discussion (https://github.com/rust-lang/rust/issues/50646#issuecomment-388596412) about making a point release with a few backports. This issue tracks the PRs we would like to backport, and if we should do the point release at all.

cc @rust-lang/release @rust-lang/core

---

* #49705: Filter out missing components from manifests
  * Reason: rustup can't download 1.26.0 on FreeBSD #49462 
* #50656: Fix `fn main() -> impl Trait` for non-`Termination` trait
  * Reason: that syntax was stabilized by accident #50595
* _No PR yet_
  * Reason: RLS doesn't work on Windows #50646

**Comment by leodasvacas** [comment]
I'd like to propose #50694 for an eventual point release, it makes `impl Trait` work with a bunch of common traits that have a `Self` default (`impl PartialEq`, `impl Add`, ...) rather than ICE.

**Comment by sanxiyn** [comment]
#49462 affects ARM as well as FreeBSD. I think the point release is warranted.

**Comment by smmalis37** [comment]
I'd like to strongly vote in favor of a point release containing the fix for #50646.

**Comment by est31** [comment]
We should also backport #49523. Right now the distributed source of the 1.26.0 release artifacts don't contain release notes for the 1.26.0 release, which is bad.

**Comment by nrc** [comment]
We're getting quite a lot of reports of problems on the RLS repo, so I'd be in favour of a point release for the RLS. I'll put together a PR.

I'd also like to update rustfmt due to https://github.com/rust-lang-nursery/rustfmt/issues/2695

**Comment by nrc** [comment]
PR for RLS/Rustfmt: https://github.com/rust-lang/rust/pull/50794

**Comment by kennytm** [comment]
I nominate #50811 to be included.

**Comment by nrc** [comment]
Discussed and approved at today's core team meeting. We wanted to include RLS/Rustfmt, termination trait, NaN in consts, and missing manifests (they were all decided either to be high reward or low risk enough to justify inclusion).

**Comment by nrc** [comment]
> Did #50694 get discussed? It's already accepted and merged for beta, got 3 issue reports and the patch was minor.

No. We missed that one. Ping @rust-lang/core - how do we feel about another inclusion?

**Comment by nrc** [comment]
iiuc it is not a stability concern, just a bug-fix since we currently stack overflow. It looks like a non-trivial patch to me, but I'm not in a great place to judge. Seems like we could live without it.

**Comment by eedahl** [comment]
Added some recent errors to #50646, currently happening on nightly, jic they're of relevance.

**Comment by sinesc** [comment]
Is this testable via rustup or download?

**Comment by pietroalbini** [comment]
There is no build yet, after the PR will be merged a build will appear in the dev-static bucker.

**Comment by Mark-Simulacrum** [comment]
Release artifacts are built, you can run `RUSTUP_DIST_SERVER=https://dev-static.rust-lang.org rustup update stable` to test it locally -- please report problems at https://internals.rust-lang.org/t/rust-1-26-1-prerelease-testing/ or here.
